packages/manager-react-components/src/hooks/datagrid/useIcebergV6.tsx

```diff
@@ -24,46 +24,35 @@ export function useResourcesIcebergV6<T = unknown>({
   queryKey,
   defaultSorting = undefined,
 }: IcebergFetchParamsV6 & IcebergV6Hook) {
-  const [totalCount, setTotalCount] = useState(0);
-  const [flattenData, setFlattenData] = useState<T[]>([]);
   const [queryParams, setQueryParams] = useState<QueryParams>({
     pageIndex: 1,
     sorting: { desc: false, id: '' },
   });
   const [sorting, setSorting] = useState<ColumnSort>(defaultSorting);
 
-  const {
-    data,
-    fetchNextPage,
-    isError,
-    isLoading,
-    error,
-    status,
-  } = useInfiniteQuery({
-    initialPageParam: null,
-    queryKey: [...queryKey, sorting],
-    queryFn: ({ pageParam }) =>
-      fetchIcebergV6<T>({
-        route,
-        pageSize,
-        page: pageParam,
-        sortBy: sorting?.id || null,
-        sortReverse: sorting?.desc,
-      }),
-    staleTime: Infinity,
-    retry: false,
-    getNextPageParam: () => queryParams.pageIndex,
-  });
+  const { data, fetchNextPage, isError, isLoading, error, status } =
+    useInfiniteQuery({
+      initialPageParam: null,
+      queryKey: [...queryKey, queryParams.pageIndex, sorting],
+      queryFn: ({ pageParam }) =>
+        fetchIcebergV6<T>({
+          route,
+          pageSize,
+          page: pageParam,
+          sortBy: sorting?.id || null,
+          sortReverse: sorting?.desc,
+        }),
+      staleTime: Infinity,
+      retry: false,
+      getNextPageParam: () => queryParams.pageIndex,
+    });
 
-  useEffect(() => {
-    const flatten = data?.pages.map((page) => page.data).flat();
-    setFlattenData(flatten);
+  const flattenData = useMemo(() => {
+    return data?.pages.map((page) => page.data).flat() ?? [];
```

Comment:
I think we can remove the useEffect and use the function "select" from tanstack query for flat the result.